src/slangPrinter.ts

```diff
@@ -1,29 +1,9 @@
-import { prettierVersionSatisfies } from './slang-utils/prettier-version-satisfies.js';
 import { isBlockComment } from './slang-utils/is-comment.js';
 import { locEnd, locStart } from './slang-utils/loc.js';
 
 import type { AstPath, Doc, ParserOptions } from 'prettier';
-import type { AstNode, StrictAstNode } from './slang-nodes';
-import type { PrintFunction } from './types';
-
-function once<T>(factory: () => T): () => T {
-  let value: T;
-  return (): T => {
-    if (typeof value === 'undefined') {
-      value = factory();
-    }
-    return value;
-  };
-}
-
-const prettierVersionCheck = once((): boolean => {
-  if (!prettierVersionSatisfies('>=2.3.0')) {
-    throw new Error(
-      'The version of prettier in your node-modules does not satisfy the required ">=2.3.0" constraint. Please update the version of Prettier.'
```

Comment:
If this PR is explicitly dropping support for v2, I wonder if this should be kept, specifying `3.0.0` instead?